epitator/structured_incident_annotator.py

```diff
@@ -0,0 +1,287 @@
+                    __last_date_mentioned=__last_date_mentioned)))
+
+        incidents = []
+        for table in tables:
+            for row_idx, row in enumerate(table.rows):
+                row_incident_date = table.metadata.get('__last_date_mentioned')
+                row_incident_location = table.metadata.get('__last_geoname_mentioned')
+                row_incident_base_type = None
+                row_incident_status = None
+                row_incident_aggregation = None
+                for column, value in zip(table.column_definitions, row):
+                    if not value: continue
+                    if column['type'] == 'date':
+                        row_incident_date = value
+                    elif column['type'] == 'geoname':
+                        row_incident_location = value
+                    elif column['type'] == 'incident_type':
+                        if "case" in value.text.lower():
+                            row_incident_base_type = "caseCount"
+                        elif "death" in value.text.lower():
+                            row_incident_base_type = "deathCount"
+                    elif column['type'] == 'incident_status':
+                        row_incident_status = value.text
+
+                row_incidents = []
+                for column, value in zip(table.column_definitions, row):
+                    if not value: continue
+                    if column['type'] == "number":
+                        column_name = column.get('name', '').lower()
+                        if row_incident_base_type:
+                            incident_base_type = row_incident_base_type
+                        else:
+                            if "cases" in column_name:
+                                incident_base_type = "caseCount"
+                            elif "deaths" in column_name:
+                                incident_base_type = "deathCount"
+                            else:
+                                incident_base_type = "caseCount"
+                        if row_incident_status:
+                            count_status = row_incident_status
+                        else:
+                            if "suspect" in column_name:
+                                count_status = "suspected"
+                            elif "confirmed" in column_name:
+                                count_status = "confirmed"
+                            else:
+                                count_status = None
+                        incident_aggregation = None
+                        if row_incident_aggregation is not None:
+                            incident_aggregation = row_incident_aggregation
+                        elif "total" in column_name:
+                            incident_aggregation = "cumulative"
+                        elif "new" in column_name:
+                            incident_aggregation = "incremental"
```

Comment:
Assuming that each row is a count of events (in the generic "event extraction" literature use of the word): In `cumulative`, each successive row represents the total of the above row's count of events plus any new events, and in `incremental` each row represents a temporally exclusive count of events?

Reply:
Cumulative represents a count of all cases with an unspecified start date and incremental represents the new cases between two known dates.

Reply:
Ok. I think that this doesn't quite map on to the way those values work in tables — because in each type of table you have the same temporal info — it's just a different value type that is being given for that time interval.

For example:

date | total cases
-----|-----
2018-04-30 | 4
2018-05-02 | 10
2018-05-02 | 15

date | new cases
-----|-----
2018-04-30 | 4
2018-05-02 | 6
2018-05-02 | 5

You have the same underlying information, and the same specificity of dates, but the values in those columns are of different types. How would you propose (and how does the annotator think about) those different cases?

Reply:
The new cases will be assigned to date ranges based on the period between row dates. The cumulative cases would not be given date ranges, only the ending date from the row would be recorded. This would result in differences in resolved cases if we found a cumulative count from another source between the dates in the table. Depending on the value of the added cumulative count, the case rates before and after it would be different. I haven't seen any tables with repeated dates so I didn't plan on handling that case.